### PR TITLE
refactor : Registered error handlers and changed return statements to raise

### LINKED ISSUE #6665: Register error handler for custom errors and change return statements to raise
https://github.com/fossasia/open-event-server/blob/9ba29f50ff2907e911dba2a51661015183ca658a/app/api/helpers/errors.py#L62-L75

https://flask.palletsprojects.com/en/1.1.x/errorhandling/#generic-exception-handlers

### PR DESCRIPTION
**Pull request by Rits1272**
Fixes #6665 

#### Short description of what this resolves:
Changed return statements to raise

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by Rits1272** [comment]
Please review @kushthedude

**Comment by kushthedude** [comment]
Please see the issue description completely

**Comment by Rits1272** [comment]
Oh Yes, working on it.

File: app/api/custom/orders.py [hunk 1]
```diff
@@ -87,7 +87,12 @@ def calculate_order_amount_wrapper(data):
     if 'discount-code' in data:
         discount_code_id = data['discount-code']
         discount_code = safe_query(db, DiscountCode, 'id', discount_code_id, 'id')
+<<<<<<< HEAD
```
Comment (niranjan94) [inline app/api/custom/orders.py]: Codacy Issue found: [Possible SCM diff in code](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4712256)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4712256&bid=15738251)

File: app/instance.py [hunk 2]
```diff
@@ -261,5 +261,53 @@ def internal_server_error(error):
...
+            'title': 'Unprocessable Entity'
+        }),
+        422,
+        {
+            'Content-Type': 'application/vnd.api+json'
+        })
+
+
+@app.errorhandler(400)
+def badRequestError_handler(error):
```
Comment (niranjan94) [inline app/instance.py]: Codacy Issue found: [Function name "badRequestError_handler" doesn't conform to '[a-z_][a-z0-9_]{1,50}$' pattern](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4712256)

File: app/instance.py [hunk 3]
```diff
@@ -261,5 +261,53 @@ def internal_server_error(error):
                          {'Content-Type': 'application/vnd.api+json'})
 
 
+@app.errorhandler(422)
+def unprocessableEntityError_handler(error):
```
Comment (niranjan94) [inline app/instance.py]: Codacy Issue found: [Function name "unprocessableEntityError_handler" doesn't conform to '[a-z_][a-z0-9_]{1,50}$' pattern](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4712256)

File: app/api/custom/orders.py [hunk 4]
```diff
@@ -87,7 +87,12 @@ def calculate_order_amount_wrapper(data):
...
+        if not TicketingManager.match_discount_quantity(discount_code, tickets, None):
```
Comment (niranjan94) [inline app/api/custom/orders.py]: Codacy Issue found: [unexpected indentation](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4712256)

File: app/instance.py [hunk 5]
```diff
@@ -261,5 +261,53 @@ def internal_server_error(error):
...
+            'title': 'Bad Request'
+        }),
+        400,
+        {
+            'Content-Type': 'application/vnd.api+json'
+        })
+
+
+@app.errorhandler(403)
+def forbiddenError_handler(error):
```
Comment (niranjan94) [inline app/instance.py]: Codacy Issue found: [Function name "forbiddenError_handler" doesn't conform to '[a-z_][a-z0-9_]{1,50}$' pattern](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4712256)

File: app/instance.py [hunk 6]
```diff
@@ -261,5 +261,53 @@ def internal_server_error(error):
...
+            'title': 'Access Forbidden'
+        }),
+        403,
+        {
+            'Content-Type': 'application/vnd.api+json'
+        })
+
+
+@app.errorhandler(404)
+def notFoundError_handler(error):
```
Comment (niranjan94) [inline app/instance.py]: Codacy Issue found: [Function name "notFoundError_handler" doesn't conform to '[a-z_][a-z0-9_]{1,50}$' pattern](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4712256)

File: app/api/custom/orders.py [hunk 7]
```diff
@@ -87,7 +87,12 @@ def calculate_order_amount_wrapper(data):
...
+            raise UnprocessableEntityError({'source': 'discount-code'}, 'Discount Usage Exceeded').respond()
+=======
     return tickets, discount_code
+>>>>>>> 3ea38141a731cf51a3c8314e02b5ec38d0b8e932
```
Comment (niranjan94) [inline app/api/custom/orders.py]: Codacy Issue found: [Possible SCM diff in code](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=4712256)

**Pull request closed without merging**